Networking/Networking/Remote/PaymentRemote.swift

```diff
@@ -56,12 +66,45 @@ public class PaymentRemote: Remote, PaymentRemoteProtocol {
             // Necessary to create a persistent cart for later checkout, the default value is `true`.
             "temporary": false
         ]
-        let request = DotcomRequest(wordpressApiVersion: .mark1_1, method: .post, path: path, parameters: parameters)
-        let response: CreateCartResponse = try await enqueue(request)
+        let response: CreateCartResponse = try await createCart(siteID: siteID, parameters: parameters)
         guard response.products.contains(where: { $0.productID == productID }) else {
             throw CreateCartError.productNotInCart
         }
     }
+
+    public func createCart(siteID: Int64, domain: PaidDomainSuggestion) async throws -> CartResponse {
+        let parameters: [String: Any] = [
+            "products": [
+                [
+                    "product_id": domain.productID,
+                    "volume": 1,
+                    "meta": domain.name,
+                    "extra": [
+                        "privacy": domain.supportsPrivacy
+                    ]
+                ]
+            ],
+            // Necessary to create a persistent cart for later checkout, the default value is `true`.
+            "temporary": false
+        ]
+        let response: CartResponse = try await createCart(siteID: siteID, parameters: parameters, encoding: JSONEncoding.default)
+
+        // Casting the values of `[String: Any]` to fixed-size integer types like `Int64` results in `nil`.
+        // https://stackoverflow.com/questions/36786883/swift-cast-any-object-to-int64-nil
+        guard let productsInCart = response["products"]?.value as? [[String: Any]],
+              productsInCart.contains(where: { ($0["product_id"] as? Int) == Int(domain.productID) }) else {
+            throw CreateCartError.productNotInCart
+        }
+        return response
+    }
+}
+
+private extension PaymentRemote {
+    func createCart<T: Decodable>(siteID: Int64, parameters: [String: Any], encoding: ParameterEncoding = URLEncoding.default) async throws -> T {
```

Comment:
Good question - the cart response might be the same, but the fields required for each use case can be very different. When creating a cart with a WPCOM plan for store creation, only the products field is used to verify the product is added successfully. However, when creating a cart with a domain when the site has a domain credit, the whole cart response is required for subsequent domain credit redemption. Since we need the whole cart response with fields that can be changed on the backend anytime, I didn't want to create a `Decodable` struct with predefined fields and thus I made the return value type generic to allow easier cart products verification in use cases that don't require the full cart response.